docs/sphinx/formats/pattern-file.txt

```diff
@@ -0,0 +1,98 @@
+Grouping files using a pattern file
+===================================
+
+Individual files can be grouped together into a single fileset using a pattern
+file.  This works for any single-file format that Bio-Formats supports, as
+long as all files are in the same format.  It is most useful for sets of TIFF,
+JPEG, PNG, etc. files that do not have any associated metadata.
+
+All files to be grouped together should be in the same folder.  The pattern
+file should be in the same folder as the other files; it can have any name,
+but must have the `.pattern` extension.  The pattern file is what must be
+opened or imported, so it may be helpful to give it a descriptive or
+easily-recognizable name.
+
+The pattern file contains a single line of text that is specially formatted to
+describe how the files should be grouped.  The file can be created in any text
+editor.
+
+The text in the pattern file can take one of several forms.  To illustrate,
+consider a folder with the following file names:
+
+::
+
+  red.tiff
+  green.tiff
+  blue.tiff
+  test_Z0_C0.png
+  test_Z1_C0.png
+  test_Z0_C1.png
+  test_Z1_C1.png
+  test_Z0_C2.png
+  test_Z1_C2.png
+  test_Z00.tiff
+  test_Z01.tiff
+
+A pattern file that groups `red.tiff`, `green.tiff`, and `blue.tiff` in that
+order would look like:
+
+::
+
+  <red,green,blue>.tiff
+
+A pattern that groups test_Z0_C0.png, test_Z1_C0.png, test_Z0_C2.png, and
+test_Z1_C2.png:
+
+::
+
+  test_Z<0-1>_C<0-2:1>.png
```

Comment:
This may need more explanation: I would have expected `C<0-2:2>` or `C<0,2>` to work, but I expect `<0-2:1>` to go from 0 to 2 by step 1.